From a7405fc1fd9a9734836a1b84e3d2a60a0eda86dc Mon Sep 17 00:00:00 2001
From: Venkatesan Mahalingam <venkatesan_mahalinga@dell.com>
Date: Tue, 8 Jun 2021 18:05:47 -0700
Subject: [PATCH] Add AFT fields for additional features.

Signed-off-by: Venkatesan Mahalingam <venkatesan_mahalinga@dell.com>
---
 common/openconfig-aft-common.yang | 41 ++++++++++++++++++++++++++++++++++++++-
 1 file changed, 40 insertions(+), 1 deletion(-)

diff --git a/common/openconfig-aft-common.yang b/common/openconfig-aft-common.yang
index 7c94e5c..f6be644 100644
--- a/common/openconfig-aft-common.yang
+++ b/common/openconfig-aft-common.yang
@@ -22,7 +22,13 @@ submodule openconfig-aft-common {
     "Submodule containing definitions of groupings that are re-used
     across multiple contexts within the AFT model.";
 
-  oc-ext:openconfig-version "0.3.2";
+  oc-ext:openconfig-version "0.3.3";
+
+  revision "2021-06-06" {
+    description
+      "Add additional fields for next-hop.";
+    reference "0.3.3";
+  }
 
   revision "2018-11-21" {
     description
@@ -128,6 +134,32 @@ submodule openconfig-aft-common {
         within the AFT. Traffic is distributed across the set of next-hops
         within the next-hop group according to the weight.";
     }
+
+    leaf origin-protocol {
+      type identityref {
+        base "oc-pol-types:INSTALL_PROTOCOL_TYPE";
+      }
+      description
+        "The protocol from which the AFT entry was learned.";
+    }
+
+    leaf distance {
+      type uint32;
+      description
+        "Admin distance for this route";
+    }
+
+    leaf metric {
+      type uint32;
+      description
+        "Distance metric for this route";
+    }
+
+    leaf uptime {
+      type string;
+      description
+        "Time since route is configured or learned in RIB";
+    }
   }
 
   grouping aft-common-entry-nexthop-config {
@@ -196,6 +228,13 @@ submodule openconfig-aft-common {
       description
         "The protocol from which the AFT entry was learned.";
     }
+
+    leaf directly-connected {
+      type boolean;
+      default false;
+      description
+        "Defines whether the nexthop is directly connected.";
+    }
   }
 
   grouping aft-common-ip-state {
-- 
2.7.4

