From 03af851b4f1a3024dff83b5e5865895ddfc6ebe8 Mon Sep 17 00:00:00 2001
From: Bing Sun <Bing_Sun@dell.com>
Date: Tue, 15 Jun 2021 10:30:54 -0700
Subject: [PATCH] Add NTP additional features

---
 openconfig-system.yang | 149 +++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 149 insertions(+)

diff --git a/openconfig-system.yang b/openconfig-system.yang
index fd05a3f..d9709e4 100644
--- a/openconfig-system.yang
+++ b/openconfig-system.yang
@@ -19,6 +19,7 @@ module openconfig-system {
   import openconfig-procmon { prefix oc-proc; }
   import openconfig-alarms { prefix oc-alarms; }
   import openconfig-messages { prefix oc-messages; }
+  import openconfig-interfaces { prefix "oc-if"; }
 
   // meta
   organization "OpenConfig working group";
@@ -119,6 +120,18 @@ module openconfig-system {
       "MD5 encryption method";
   }
 
+  identity NTP_AUTH_SHA1 {
+    base oc-sys:NTP_AUTH_TYPE;
+    description
+      "SHA1 encryption method";
+  }
+
+  identity NTP_AUTH_SHA2_256 {
+    base oc-sys:NTP_AUTH_TYPE;
+    description
+      "SHA2-256 encryption method";
+  }
+
   // typedef statements
 
   typedef timezone-name-type {
@@ -523,6 +536,42 @@ module openconfig-system {
         "Indicates whether this server should be preferred
         or not.";
     }
+
+    leaf key-id {
+      type uint16;
+      description
+        "Integer identifier used by the client and server to
+        designate a secret key. The client and server must use
+        the same key id.";
+    }
+
+    leaf minpoll {
+      type uint8 {
+        range "3..17";
+      } 
+      default 6;
+
+      description
+        "The minimum poll interval used in this association.
+        Default is 6.";
+        reference
+        "RFC 5905: Network Time Protocol Version 4: Protocol and
+        Algorithms Specification";
+    }
+
+    leaf maxpoll {
+      type uint8 {
+        range "3..17";
+      }
+      default 10;
+
+      description
+        "The maximum poll interval used in this association.
+        Default is 10.";
+        reference
+        "RFC 5905: Network Time Protocol Version 4: Protocol and
+        Algorithms Specification";
+    }
   }
 
   grouping system-ntp-server-state {
@@ -590,6 +639,79 @@ module openconfig-system {
       description
         "Polling interval of the peer";
     }
+
+    leaf peerdelay {
+      type decimal64 {
+        fraction-digits 3;
+      }
+      description
+        "The round-trip delay to the server using type decimal64, in milliseconds";
+    }
+
+    leaf peeroffset {
+      type decimal64 {
+        fraction-digits 3;
+      }
+      description
+        "Estimate of the current time offset from the peer.  This is
+        the time difference between the local and reference clock.
+        Using type decimal64, in milliseconds.";
+    }
+
+    leaf peerjitter {
+      type decimal64 {
+        fraction-digits 3;
+      }
+      description
+        "The mean deviation of multiple time samples from this peer.";
+    }
+
+    leaf selmode {
+      type string {
+        length "1";
+      }
+      description
+        "Character before the peer address or peer name:
+        * Synchronized to this peer
+        # Almost synchronized to this peer
+        + Peer selected for possible synchronization
+        - Peer is a candidate for selection
+        ~ Peer is statically configured.";
+    }
+
+    leaf refid {
+      type oc-inet:host;
+      description
+        "Reference clock type or address for the peer.";
+    }
+ 
+    leaf peertype {
+      type string {
+        length "1";
+      }
+      description
+        "Type of the peer.
+        u: unicast or manycast client
+        b: broadcast or multicast client
+        l: local reference clock
+        s: symmetric peer
+        A: manycast server
+        B: boadcast server
+        M: multicast server.";
+    }
+
+    leaf now {
+      type uint32;
+      description
+        "Indicates the time sine the NTP packet was not received or
+        last synchronized, in seconds.";
+    }
+
+    leaf reach {
+      type string;
+      description
+        "An bitmap of size 8 bits in octal string indicates the status of the last eight NTP messages.";
+     }
   }
 
   grouping system-ntp-server-top {
@@ -664,6 +786,13 @@ module openconfig-system {
       description
         "NTP authentication key value";
     }
+
+    leaf encrypted {
+      type boolean;
+      default false;
+      description
+        "NTP authentication key-value is encrypted";
+    }
   }
 
   grouping system-ntp-auth-keys-state {
@@ -740,6 +869,26 @@ module openconfig-system {
         system will only use packets containing a trusted
         authentication key to synchronize the time.";
     }
+
+    leaf-list trusted-key {
+      type uint16;
+      description
+        "Integer identifier used by the client and server to
+        designate a secret key.  The client and server must use
+        the same key id.";
+    }
+
+    leaf-list ntp-source-interface {
+      type oc-if:base-interface-ref;
+      description
+        "NTP source interface name";
+    }
+
+    leaf vrf {
+      type string;
+      description
+        "The VRF instance NTP is enabled in.";
+    }
   }
 
   grouping system-ntp-state {
-- 
2.7.4

